addon/components/pie-chart.js

```diff
@@ -483,9 +483,17 @@ export default ChartComponent.extend(FloatingTooltipMixin,
     groups.select('path').attr(this.get('sliceAttrs'));
     
     var labelWidth = this.get('labelWidth');
+    var maxLabelWidth = this.get('outerWidth')/2 - this.get('labelPadding');
     var labelTrimmer = LabelTrimmer.create({
-      getLabelSize: function() {
-        return labelWidth;
+      getLabelSize: function(d, selection) {
```

Comment:
`d` is just the data of the current element. It's been commonly used in the pie-charts.js as the variable so I just used the same.